pokemongo_bot/cell_workers/evolve_pokemon.py

```diff
@@ -81,23 +78,25 @@ def work(self):
                 if pokemon.can_evolve_now():
                     self._execute_pokemon_evolve(pokemon, cache)
 
-    def _print_check(self, has, needs):
-        self.emit_event(
-            'pokemon_evolve_check',
-            formatted='Evolvable: {has}/{need}',
-            data={
-                'has': has,
-                'needs': needs
-            }
-        )
+    def _log_interval_if_should(self, has, needs):
+        self._compute_next_log_update()
+        if self._should_log_update:
+            self.emit_event(
+                'pokemon_evolve_check',
+                formatted='Evolvable: {has}/{need}',
+                data={'has': has, 'needs': needs}
+            )
```

Comment:
no, it may return either `True` or `False`,
but I have just realized I made it impossible for `self.next_log_update` to be `None`. Nice catch. 👍